actionpack/test/journey/router/utils_test.rb

```diff
@@ -42,6 +42,22 @@ def test_normalize_path_maintains_string_encoding
         def test_normalize_path_with_nil
           assert_equal "/", Utils.normalize_path(nil)
         end
+
```

Comment:
These tests pass regardless of your change. I think you should add one to `actionpack/test/dispatch/routing_test.rb` instead. Try this:

```ruby
def test_deeply_scoped_root_unscoped_access
  draw do
    scope "(:locale)" do
      scope "(:platform)" do
        scope "(:browser)" do
          root to: "projects#index"
        end
      end
    end
  end

  assert_equal "/", root_path
  get "/"
  assert_equal "projects#index", @response.body
end
```